From ecf4324038220e1c0cdc1f3498eb53e10a6eb97a Mon Sep 17 00:00:00 2001
From: Andrei Warkentin <andrei.warkentin@intel.com>
Date: Fri, 17 Feb 2023 18:57:22 -0600
Subject: [PATCH 6/8] Dxe: add RISCV64 to mMachineTypeInfo

This fixes messages like:
"Image type AARCH64 can't be loaded on <Unknown> UEFI system"

Cc: Sunil V L <sunilvl@ventanamicro.com>
Cc: Daniel Schaefer <git@danielschaefer.me>
Cc: Liming Gao <gaoliming@byosoft.com.cn>
Cc: Jian J Wang <jian.j.wang@intel.com>
Signed-off-by: Andrei Warkentin <andrei.warkentin@intel.com>
---
 MdeModulePkg/Core/Dxe/Image/Image.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/MdeModulePkg/Core/Dxe/Image/Image.c b/MdeModulePkg/Core/Dxe/Image/Image.c
index 06cc6744b8c6..8704ebea9a7c 100644
--- a/MdeModulePkg/Core/Dxe/Image/Image.c
+++ b/MdeModulePkg/Core/Dxe/Image/Image.c
@@ -81,7 +81,8 @@ GLOBAL_REMOVE_IF_UNREFERENCED MACHINE_TYPE_INFO  mMachineTypeInfo[] = {
   { EFI_IMAGE_MACHINE_IA64,           L"IA64"    },
   { EFI_IMAGE_MACHINE_X64,            L"X64"     },
   { EFI_IMAGE_MACHINE_ARMTHUMB_MIXED, L"ARM"     },
-  { EFI_IMAGE_MACHINE_AARCH64,        L"AARCH64" }
+  { EFI_IMAGE_MACHINE_AARCH64,        L"AARCH64" },
+  { EFI_IMAGE_MACHINE_RISCV64,        L"RISCV64" },
 };
 
 UINT16  mDxeCoreImageMachineType = 0;
-- 
2.25.1

